### PR TITLE
Better CI

### LINKED ISSUE #16: Better CI
Rust and TOML formatting as well as spellcheck up to `Forest` standards

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- added Makefile for all actions, currently only lint
- added lint-all workflow

**Reference issue to close (if applicable)**

Closes #16 

**Other information and links**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
+
+install-deps:
+	apt-get update -y
+	apt-get install --no-install-recommends -y build-essential clang protobuf-compiler ocl-icd-opencl-dev aria2 cmake
```
Comment (lemmih) [inline Makefile:4]: These dependencies should be updated.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+lint-clippy:
+	cargo clippy
+
+# Formats Rust and TOML files
+fmt:
+	cargo fmt --all
+	taplo fmt
+
+clean:
+	@echo "Cleaning local packages..."
```
Comment (lemmih) [inline Makefile:41]: Why isn't `cargo clean` sufficient?

**Review by lemmih — CHANGES_REQUESTED** [review]
Let's drop the spellcheck.

File: Makefile (hunk 1 above)
Comment (tyshko5) [reply Makefile:4]: These are the same ones we use in Forest. Which of them should be updated?

Comment (lemmih) [reply Makefile:4]: Are all of them required by these crates?

File: Makefile (hunk 2 above)
Comment (tyshko5) [reply Makefile:41]: Fixed

File: Makefile (hunk 1 above)
Comment (tyshko5) [reply Makefile]: removed them

**Comment by tyshko5** [comment]
> Let's drop the spellcheck.

Why?

**Comment by lemmih** [comment]
> > Let's drop the spellcheck.
> 
> Why?

We're just vendoring the builtin-actors. Since patches aren't pushed upstream, we would have to fix the same spelling/formatting errors over and over again for each new release.

**Comment by tyshko5** [comment]
> > > Let's drop the spellcheck.
> > 
> > 
> > Why?
> 
> We're just vendoring the builtin-actors. Since patches aren't pushed upstream, we would have to fix the same spelling/formatting errors over and over again for each new release.

But as far as I understand, this concerns only the newest (currently) v10 actors, but not anything older, right?

**Comment by lemmih** [comment]
> > > > Let's drop the spellcheck.
> > > 
> > > 
> > > Why?
> > 
> > 
> > We're just vendoring the builtin-actors. Since patches aren't pushed upstream, we would have to fix the same spelling/formatting errors over and over again for each new release.
> 
> But as far as I understand, this concerns only the newest (currently) v10 actors, but not anything older, right?

Not sure what you mean. Version 11 of the actors will be a modified copy of version 10. Each version of the actors are a slightly modified copy of the previous version.

**Comment by tyshko5** [comment]
> > > > > Let's drop the spellcheck.
> > > > 
> > > > 
> > > > Why?
> > > 
> > > 
> > > We're just vendoring the builtin-actors. Since patches aren't pushed upstream, we would have to fix the same spelling/formatting errors over and over again for each new release.
> > 
> > 
> > But as far as I understand, this concerns only the newest (currently) v10 actors, but not anything older, right?
> 
> Not sure what you mean. Version 11 of the actors will be a modified copy of version 10. Each version of the actors are a slightly modified copy of the previous version.

I mean that there is no issue with keeping the fixes for the older releases.

**Comment by lemmih** [comment]
> I mean that there is no issue with keeping the fixes for the older releases.

Fixing the same formatting error over and over again every three months would be a waste of time.

**Comment by tyshko5** [comment]
> > I mean that there is no issue with keeping the fixes for the older releases.
> 
> Fixing the same formatting error over and over again every three months would be a waste of time.

Removed spellcheck

File: .config/fil_actors.dic [hunk 3]
```diff
@@ -0,0 +1,93 @@
+92
```
Comment (lemmih) [inline .config/fil_actors.dic:1]: The two dictionaries are no longer necessary.

**Review by lemmih — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**